### DISCUSSION TITLE
Beta release v1.2.0 of DynAdjust

### DISCUSSION THREAD

**Discussion by rogerfraser** [GeoscienceAustralia/DynAdjust#149] [Archive]
# Overview
This beta release contains several code enhancements and bug fixes, and addresses issues #15, #30, #78, #81, #82, and #85.  The following summarises the related (major) pull requests associated with these code changes.

Please provide feedback or comments on issues experienced with this beta release in the discussion (rather than creating a new issue).

## General code enhancements
- Numerous (too many to list) code enhancements have been applied to address issues identified by [Codacy](https://app.codacy.com/gh/icsm-au/DynAdjust/dashboard) and code coverage identified by [Coveralls](https://coveralls.io/github/icsm-au/DynAdjust) and [Codecov](https://codecov.io/gh/icsm-au/DynAdjust).

## Enhancements and bug fixes for **import**
- See #87 for changes to permit the sorting of measurements in the m2s file
- See #98 for enhancements to the application of a GNSS baseline scalars file
- In relation to extending discontinuity renaming (#15):
   - See #88 for the initial changes to **import** to address this issue
   - See #147 for final changes to **import** and **adjust** to complete the work.
   - Note that the changes applied in #147 change the structure of information written to the binary station file, meaning that `*.bst` files created by earlier versions of **import** cannot be reused by this or future versions of DynAdjust.

## Enhancements and bug fixes for **geoid**
- See #89, #90, #94, #108, #109 and #146 for code changes, bug fixes and test data

## Enhancements to **adjust**
- See #83 for enhancements to address column overrun identified in issue #78
- See #84 for a fix to incorrect iteration count identified in #82
- See #86 for changes to prevent output of unused stations to the adjusted stations file, addressing #85
- See #87 for changes to permit the sorting of the measurements to station table by measurement count
- See #92 for changes to provide a capability to export adjusted station coordinates and uncertainties as a GNSS Y cluser
- See #99 for an enhancement that prints memory consumption to the `.adj` and `.dbg` files on verbose execution
- See #107 for a fix to a defect in the way adjusted measurements are printed on each iteration in staged mode
- See comments above in import (#147)

## Enhancements to **plot** and upgrade to GMT6
- See #137 for a fix to a defect in plotting single blocks
- See #139 for code changes addressing #30

## DynAdjust User's Guide
- See #138 and #146 for the updates applied to the User's Guide

<hr /><em>This discussion was created from the release <a href='https://github.com/icsm-au/DynAdjust/releases/tag/v1.2-beta.0'>Beta release v1.2.0 of DynAdjust</a>.</em>

**Comment by harry093** [comment]
@rogerfraser I have done some testing and came across the following error

- Error: ALIC_1900139 is referenced in measurement 'X' SS23137 - ALIC_1900139
  but is not in the stations map.  
  Please ensure that ALIC_1900139 is included in the list of stations.

I believe that ALIC_1900139 should be ALIC_1996080.

The attached zip file contains everything needed to reproduce what I am doing. It's a good test case because it's small and uses almost all the new capability - renaming stations and taking that into account when listing stations to include and stations to hold fixed. I have not tested excluding stations. The files in the directory old/ are how I used to do this without the new capability.

If you have any questions feel free to contact me.

[cluster_09139.zip](https://github.com/icsm-au/DynAdjust/files/6788174/cluster_09139.zip)

**Comment by rogerfraser** [comment]
Thanks @harry093 for raising this and providing a sample data set.  This looks quite similar to an issue @BatchelorJ found, which I hope can be addressed in the release.
